src/cgns_f.F90

```diff
@@ -4638,7 +4644,15 @@ END FUNCTION cgp_poly_section_write
     END INTERFACE
 
     c_name = TRIM(sectionname)//C_NULL_CHAR
-    ier = cgp_poly_section_write(fn, B, Z, c_name, type, start, end, maxoffset, nbndry, S)
+    c_fn = INT(fn, c_int)
+    c_B = INT(B, c_int)
+    c_Z = INT(Z, c_int)
+    c_nbndry = INT(nbndry, c_int)
```

Comment:
No need to create extra variables for intent in variables. Also just do ier = INT( cpg...,)